ada_feeding_perception/ada_feeding_perception/food_on_fork.py

```diff
@@ -0,0 +1,363 @@
+                                    "to the fork tine)",
+                        read_only=True,
+                    ),
+                ),
+                (
+                    "max_dist",
+                    None,
+                    ParameterDescriptor(
+                        name="max_dist",
+                        type=ParameterType.PARAMETER_INTEGER,
+                        description="maximum depth to consider (note that 330 is approx distance "
+                                    "to the fork tine)",
+                        read_only=True,
+                    ),
+                ),
+                (
+                    "model_location",
+                    None,
+                    ParameterDescriptor(
+                        name="model_location",
+                        type=ParameterType.PARAMETER_STRING,
+                        description="model location",
+                        read_only=True,
+                    ),
+                ),
+                (
+                    "test",
+                    None,
+                    ParameterDescriptor(
+                        name="test",
+                        type=ParameterType.PARAMETER_BOOL,
+                        description="testing",
+                        read_only=True,
+                    ),
+                ),
+            ],
+        )
+
+    def normalize_to_uint8(self, img):
+        """
+        Normalize the image to 0-255
+
+        Parameters:
+        ----------
+        img: image to normalize
+        """
+        img_normalized = ((img - img.min()) / (img.max() - img.min()) * 255).astype(
+            "uint8"
+        )
+        return img_normalized
+
+    def listener_callback_depth(self, depth_img_msg: Image) -> None:
+        """
+        Calculates and publishes probability indicating whether the provided depth image has food
+        on fork
+
+        Parameters:
+        ----------
+        depth_img_msg: depth image message from the subscriber
+        """
+        depth_img = None
+        try:
+            depth_img = self.bridge.imgmsg_to_cv2(depth_img_msg, "passthrough")
+        except CvBridgeError as e:
+            print(e)
+
+        if depth_img is not None:
+            # VVVV---BELOW is the code when using single feature of only the NUMBER OF PIXELS---VVVV
+            # num_pixels = self.food_on_fork_num_pixels(depth_img)
+            # if self.test:
+            #     self.get_logger().info("num pixels: " + str(num_pixels))
+            # prediction = self.predict_food_on_fork(num_pixels)
+            # if self.test:
+            #     self.get_logger().info("Prediction Probability: " + str(prediction))
+            # ^^^^---ABOVE is the code when using single feature of only the NUMBER OF PIXELS---^^^^
+
+            # VVVV---BELOW is the code when using EACH PIXEL as a feature---VVVV
+            depth_img_copy = np.copy(depth_img)
```

Comment:
Why do you need to copy it? This seems like an unnecessary computation to me.

Reply:
made some minor edits to not require the copying in all cases; it was to draw the rectangle but only needed in debugging cases.